File: java/ankit_kanyal/devnest_problems/phase_1/IntervalsNonOverlapping.java
// link- https://devsnest.in/algo-challenges/intervals-non-overlapping

static int solve(int[][] interval){
//CODE HERE 
    Arrays.sort(interval,Comparator.comparingInt(x->x[0]));
    int prevStart=interval[0][0],prevEnd=interval[0][1];
    int curStart,curEnd;
    int n=interval.length;
    int count=0;
    for(int i=1;i<interval.length;i++){
        curStart=interval[i][0];
        curEnd=interval[i][1];

        if(curStart<prevEnd){
            count++;
            prevEnd=Math.min(curEnd,prevEnd);
        }
        else prevEnd=curEnd;
        
    }
    
    return count;
}

File: Python/chandu/Intervals_Non-Overlapping.py
def solve(interval):
    # CODE HERE
    interval.sort(key = lambda x :x[1])
    print(interval)
    count = 1
    start, end  = interval[0]
    for s,e in interval:
        if s<end:
            continue
        else:
            start, end = s,e
            count += 1
    
    return len(interval) - count
